File: 1st/Hyeseung/Programmers/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        HashMap<String, Integer> map = new HashMap<>();
        
        for(int i = 0; i < name.length; i++) map.put(name[i], yearning[i]);
        
        for(int i = 0; i < photo.length; i++) {
            for(int j = 0; j < photo[i].length; j++) {
                answer[i] += (map.get(photo[i][j]) == null) ? 0 : map.get(photo[i][j]);
            }
        }
        
        return answer;
    }
}

File: 2st/Yeonwoo/추억 점수.py
def solution(name, yearning, photo):
    answer = [] 
    for i in range(len(photo)):
        temp = 0
        for j in range(len(name)):
            if name[j] in photo[i]:
                temp += yearning[j]
        answer.append(temp)
            
    return answer